main.elm

```diff
@@ -30,15 +30,23 @@ type alias Coord =
     ( Int, Int )
 
 
+{-| Dated values
+-}
+type alias Dated a =
```

Comment:
When a type is defined with only one constructor like my `Date` example, you can pattern match on it directly in function definition like:
```elm
fadingText : Time -> Dated String -> Html Msg
fadingText now (Dated time str) =
    ...
```
Small guide to destructuring (pattern matching) in elm: https://gist.github.com/yang-wei/4f563fbf81ff843e8b1e